merino/curated_recommendations/corpus_backends/corpus_api_backend.py

```diff
@@ -245,23 +245,31 @@ async def _fetch_from_backend(self, surface_id: ScheduledSurfaceId) -> list[Corp
         res.raise_for_status()
         data = res.json()
 
+        # log to Sentry if GraphQL returned errors
+        if res.status_code == 200 and "errors" in data:
+            for error in data["errors"]:
+                logger.exception(error)
```

Comment:
My bad.. I realize the Jira ticket is ambiguous about what I think should happen.

I don't think this response will have (partially) useful data if there is a GraphQL error, so I think it's better to raise an exception here, and not continue execution of the function.
```suggestion
            raise BackendError("curated-corpus-api returned GraphQL error(s) {data["errors"]}")
```
You can import `BackendError` using `from merino.exceptions import BackendError`.

In `_revalidate_cache` I would catch this BackendError along with the HTTPError, by changing `except BackendError as e:` to `except (HTTPError, BackendError) as e:`.